src/test/java/com/cloudbees/jenkins/support/impl/RunningBuildsTest.java

```diff
@@ -101,4 +109,20 @@ public void addContentsPipeline() throws Exception {
 
         assertThat(output, containsString(String.format(EXPECTED_OUTPUT_FORMAT, p.getName(), workflowRun.getNumber())));
     }
+
+    private static class SemaphoreBuilder extends TestBuilder {
+
+        private final OneShotEvent event = new OneShotEvent();
+
+        @Override
+        public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException {
+            System.err.println("Waiting for semaphore...");
```

Comment:
Why the error log?

Reply:
It resembles the logging in [SemaphoreStep](https://github.com/jenkinsci/workflow-support-plugin/blob/workflow-support-3.4/src/test/java/org/jenkinsci/plugins/workflow/test/steps/SemaphoreStep.java). When using this builder in a test, this output is easily visible next to the normal output from Jenkins. If `BuildListener` would be used, this would be harder to access.